From 81676116b8e36fe01334f456cb91fc34b34b5536 Mon Sep 17 00:00:00 2001
From: Ronan Le Martret <ronan.lemartret@iot.bzh>
Date: Wed, 28 Feb 2024 14:30:50 +0100
Subject: [PATCH] Hack to stabilize nfc reader

* In some case (unstable) we've got a problem on the acr122u reader.
 Remove the timeout on "libusb_bulk_transfer" solve the problem.
---
 src/ccid_usb.c | 2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/ccid_usb.c b/src/ccid_usb.c
index ff157a1..9858cba 100644
--- a/src/ccid_usb.c
+++ b/src/ccid_usb.c
@@ -1060,7 +1060,7 @@ read_again:
 	{
 		rv = libusb_bulk_transfer(usbDevice[reader_index].dev_handle,
 			usbDevice[reader_index].bulk_in, buffer, *length,
-			&actual_length, usbDevice[reader_index].ccid.readTimeout);
+			&actual_length, 0);
 
 		if (rv < 0)
 		{
-- 
2.43.2

